array_apply.json

```diff
@@ -34,11 +34,16 @@
                     },
                     {
                         "name": "index",
-                        "description": "The zero-based index of the current element being processed.",
+                        "description": "The zero-based index or the label of the current element being processed. Labels take preference over indices.",
```

Comment:
This is weird. Either `array_apply` should work only on plain arrays and forward index, or it should work only on labeled arrays and forward label (and maybe index). In any case it should not be one or another at the backend's discretion.

Reply:
It's not at the backend's discretion: "Labels take preference over indices."
So whenever a label is available (i.e. an array of subtype `labeled-array` has been passed) the label has to be returned instead of the array.

Reply:
I added an optional label parameter to the callbacks in array_apply and array_filter.